usmqe_tests/api/grafana/test_gluster_at_a_glance.py

```diff
@@ -39,18 +43,195 @@ def test_layout():
+        'At-a-glance': [
+            'Health',
+            'Snapshots',
+            'Hosts',
+            'Volumes',
+            'Bricks',
+            'Geo-Replication Session',
+            'Connection Trend',
+            'IOPS',
+            'Capacity Utilization',
+            'Capacity Available',
+            'Weekly Growth Rate',
+            'Weeks Remaining',
+            'Throughput Trend'],
+        'Status': [
+            'Volume Status',
+            'Host Status',
+            'Brick Status']}
+    structure = {}
+    for row in layout["dashboard"]["rows"]:
+        structure[row["title"]] = []
+        for panel in row["panels"]:
+            if panel["title"]:
+                structure[row["title"]].append(panel["title"])
+            elif "displayName" in panel.keys() and panel["displayName"]:
+                structure[row["title"]].append(panel["displayName"])
+
+    LOGGER.debug("defined layout structure = {}".format(structure_defined))
+    LOGGER.debug("layout structure in grafana = {}".format(structure))
+    pytest.check(
+        structure_defined == structure,
+        "defined structure of panels should equal to structure in grafana")
+
+
+def test_status(os_distro, cluster_reuse):
+    """@pylatest grafana/status
+    API-grafana: hosts
+    *******************
+
+    .. test_metadata:: author fbalak@redhat.com
+
+    Description
+    ===========
+
+    Check that Grafana panel *Hosts* is showing correct values.
+    """
+    gluster = GlusterCommon()
+    states = gluster.get_host_states_from_trusted_pool(
+        pytest.config.getini("usm_cluster_member"))
+    grafana = grafanaapi.GrafanaApi()
+    graphite = graphiteapi.GraphiteApi()
+    if os_distro == "Red Hat Enterprise Linux Server":
+        prefix = "webadmin"
+    else:
+        prefix = "tendrl"
+    """@pylatest grafana/hosts
+    .. test_step:: 1
+
+        Send **GET** request to:
+        ``GRAFANA/dashboards/db/PREFIX-gluster-at-a-glance``.
+
+    .. test_result:: 1
+
+        JSON structure containing data related to layout is returned.
+    """
+
+    layout = grafana.get_dashboard("{}-gluster-at-a-glance".format(prefix))
+    pytest.check(
+        len(layout) > 0,
+        layout)
+    """@pylatest grafana/hosts
+    .. test_step:: 2
+
+        Send **GET** request to ``GRAPHITE/render?target=[target]&format=json``
+        where [target] is part of uri obtained from previous GRAFANA call.
+        There should be target for Total number of hosts.
+        Compare number of hosts from Graphite with value retrieved from gluster
+        command.
+
+    .. test_result:: 2
+
+        JSON structure containing data related to Total host count with last
+        value coresponding with output of gluster command.
+    """
+    target = [
+        panel for panel in layout["dashboard"]["rows"][1]["panels"]
+        if "clusterName" in panel and panel["clusterName"] == "Hosts"
+            ][0]["targets"][0]["target"]
```

Comment:
All the indexes here are slightly confusing/not clear what it means... could you please put there some comment, explaining what it means?
It will be better for further debugging, for example if the json data structure will change (not sure if it is plausible).

Reply:
I'm not sure if this needs to be commented. It will be probably on more places and it's basically parsing of some complicated json string that developer need to look at when he/she is debugging this string. @mbukatov @ltrilety thoughts?